types/msg.go

```diff
@@ -96,6 +101,47 @@ type IBCMsg struct {
 	CloseChannel *CloseChannelMsg `json:"close_channel,omitempty"`
 }
 
+type GovMsg struct {
+	// This maps directly to [MsgVote](https://github.com/cosmos/cosmos-sdk/blob/v0.42.5/proto/cosmos/gov/v1beta1/tx.proto#L46-L56) in the Cosmos SDK with voter set to the contract address.
+	Vote VoteMsg `json:"vote,omitempty"`
+}
+
+type VoteOption int
+
+type VoteMsg struct {
+	ProposalId uint64     `json:"proposal_id"`
+	Vote       VoteOption `json:"vote"`
+}
+
+const (
+	Yes VoteOption = iota
+	No
+	Abstain
+	NoWithVeto
+)
+
+var toVoteOption = map[string]VoteOption{
+	"yes":          Yes,
+	"no":           No,
+	"abstain":      Abstain,
+	"no_with_veto": NoWithVeto,
+}
+
+func (s *VoteOption) UnmarshalJSON(b []byte) error {
+	var j string
+	err := json.Unmarshal(b, &j)
+	if err != nil {
+		return err
+	}
+
+	voteOption, ok := toVoteOption[j]
+	if !ok {
+		return fmt.Errorf("invalid vote option %v", j)
```

Comment:
Let's help to differentiate the error message and the value, especially when thinking of `j` being the empty string.

```suggestion
		return fmt.Errorf("invalid vote option '%v'", j)
```